server/client.go

```diff
@@ -4051,7 +4051,8 @@ func (c *client) processInboundClientMsg(msg []byte) (bool, bool) {
 		c.mu.Lock()
 		if c.opts.NoResponders {
 			if sub := c.subForReply(c.pa.reply); sub != nil {
-				proto := fmt.Sprintf("HMSG %s %s 16 16\r\nNATS/1.0 503\r\n\r\n\r\n", c.pa.reply, sub.sid)
+				hdrLen := 32 /* header without the subject */ + len(c.pa.subject)
+				proto := fmt.Sprintf("HMSG %s %s %d %d\r\nNATS/1.0 503\r\nNats-Subject: %s\r\n\r\n\r\n", c.pa.reply, sub.sid, hdrLen, hdrLen, c.pa.subject)
```

Comment:
I picked Nats-Subject as the header name because it is used elsewhere.

Reply:
In that case it means the original subject the current message was published as - not sure we want to use that.

Reply:
Also the only way you would get this is if you are responding and adding a reply subject - which means that when the server answers with no responders, you know what the reply subject that you published a response to is.

Reply:
Why I need the feature:

- Service A: pub reply to  MyService.Endpoint  with a _INBOX.<hash>
- Service B: sub to MyService.Endpoint  and publish multiple times (grpc server stream) to the client _INBOX.hash with a reply option to MyService.Endpoint. 
- The server tracks each inbox in a hashmap and deletes it when it receives a 503.

To resolve the issue, I'm doing something similar but my service subscribe to MyService.> and setting the reply as: MyService.Endpoint.<hash>, I also could have two subscriptions, one for MyService.Endpoint and another for MyService.Endpoint.>

Reply:
Hi, @aricart. Can we review and consider this change?